docs/config-examples: add niksingh710's config (#1708)
nix-community/nixvim: docs/user-guide/config-examples.md

| [hbjydev/hvim] | |
| [MikaelFangel/nixvim-config] | An easy-setup configuration for NixVim, focused on straightforward customization |
| [nicolas-goudry/nixvim-config] | Heavily inspired by AstroNvim |

> | [niksingh710/nvix] | Ported from the previous lazy based raw config. Contains Most of the use case. (Custom plugins/Nixpkgs plugins/Nixvim Plugins all are covered) |

Reviewer: ```suggestion
| [niksingh710/nvix] | Ported from the previous lazy based raw config. Contains most of the use cases. (Custom plugins/Nixpkgs plugins/Nixvim Plugins are all covered) |
```

> | [niksingh710/nvix] | Ported from the previous lazy based raw config. Contains most of the use cases. (Custom plugins/Nixpkgs plugins/Nixvim Plugins are all covered) |

Reviewer 2: Sorry to nitpick, I think something like this would read better?

```suggestion
| [niksingh710/nvix] | Ported from a lazy-based lua config. Contains examples of most use cases, including custom, nixpkgs, & nixvim plugins. |
```

Author: I agree that my initial attempt was somewhat verbose, resulting in a lengthy sentence. Your version is much more concise and polished.
